src/backend/glutin/mod.rs

```diff
@@ -127,21 +130,24 @@ impl Display {
     pub fn rebuild(
         &self,
         wb: glutin::WindowBuilder,
-        cb: glutin::ContextBuilder,
+        cb: glutin::ContextBuilder<NotCurrent>,
         events_loop: &glutin::EventsLoop,
     ) -> Result<(), DisplayCreationError>
     {
         // Share the display lists of the existing context.
         let new_gl_window = {
             let gl_window = self.gl_window.borrow();
             let cb = cb.with_shared_lists(gl_window.context());
-            glutin::WindowedContext::new_windowed(wb, cb, events_loop)?
+            cb.build_windowed(wb, events_loop)?
+        };
+        let new_gl_window = unsafe {
+            new_gl_window.treat_as_current()
```

Comment:
This context is not current. You are treating it as current, even though it is not. You are then placing it into the takeable.

This is not incorrect, per se, but it also fails to utilize glutin's context currency tracking system.

Reply:
@ZeGentzy yeah this port is definitely not perfect. Probably in order to do it properly, one should store a `glutin::WindowedContext<T>` everywhere but this would most likely be very involved especially for users as suddenly the glium context is taken by move and stuff.